Shared/Samples/Create and save KML file/CreateAndSaveKMLView.Model.swift

```diff
@@ -0,0 +1,135 @@
+// Copyright 2023 Esri
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import ArcGIS
+import Combine
+
+extension CreateAndSaveKMLView {
+    /// The model used to store the geo model and other expensive objects used in this view.
+    @MainActor class Model: ObservableObject {
+        /// A dark gray map.
+        let map = Map(basemapStyle: .arcGISDarkGray)
+        
+        /// The geometry editor that deals with the sketching.
+        let geometryEditor: GeometryEditor
+        
+        /// The KML style used for the geometry.
+        var kmlStyle: KMLStyle?
+        
+        /// A KML document that serves as a container for features and styles.
+        var kmlDocument = KMLDocument()
+        
+        /// A KMZ file that can be exported with the system's file exporter.
+        var kmzFile = KMZFile(document: .init())
+        
+        /// A Boolean value indicating if the geometry can be cleared from the geometry editor.
+        @Published private(set) var canClearCurrentSketch = false
+        
+        /// A Boolean value indicating if the saved sketches can be cleared.
+        @Published private(set) var canClearSavedSketches = false
+        
+        /// A Boolean value indicating if the geometry editor has started.
+        @Published var isStarted = false
+        
+        /// A Boolean value indicating if the geometry can be saved to a graphics overlay.
+        @Published private(set) var canSave = false
+        
+        /// The current geometry of the geometry editor.
+        @Published private(set) var geometry: Geometry? {
+            didSet {
+                canClearCurrentSketch = geometry.map { !$0.isEmpty } ?? false
```

Comment:
I feel like this boolean can be reversely named 😂 , as its usage is `.disabled(!model.canClearCurrentSketch)`

Maybe sth like `clearButtonDisabled` so it can become `geometry.map(\.isEmpty) ?? true`